File: backend/src/models/Resenia.ts
export default class Resenia {
  private esPremium: boolean
  private fechaResenia: Date
  private puntaje: number
  constructor(esPremium: boolean, fechaReseña: Date, puntaje: number) {
    this.esPremium = esPremium
    this.fechaResenia = fechaReseña
    this.puntaje = puntaje
  }

  obtenerPuntaje(): number {
    return this.puntaje
  }

  obtenerEsPremium(): boolean {
    return this.esPremium
  }

  esDePeriodo(fechaInicio: Date, fechaFin: Date): boolean {
    return this.fechaResenia >= fechaInicio && this.fechaResenia <= fechaFin
  }

  getFechaResenia(): Date {
    return this.fechaResenia
  }
}

File: ReporteRankingVinos/src/main/java/com/Bonvino/ReporteRankingVinos/models/Resenia.java
package com.Bonvino.ReporteRankingVinos.models;

import java.util.Date;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@NoArgsConstructor
public class Resenia {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Integer id;

  private boolean esPremium;
  private Integer puntaje;
  private Date fechaResenia;

  @ManyToOne
  @JoinColumn(name = "vino_id", nullable = false)
  private Vino vino;

  public Resenia(boolean esPremium, Integer puntaje, Date fechaResenia) {
    this.esPremium = esPremium;
    this.puntaje = puntaje;
    this.fechaResenia = fechaResenia;
  }

  public Integer obtenerPuntaje() {
    return this.puntaje;
  }

  public boolean sosDeSommelier() {
    return this.esPremium;
  }

  public boolean esDePeriodo(Date fechaDesde, Date fechaHasta) {
    return this.fechaResenia.after(fechaDesde) && this.fechaResenia.before(fechaHasta);
  }

  public Date getFechaResenia() {
    return this.fechaResenia;
  }
}
